lib/core/include/irods/rodsError.h

```diff
@@ -44,6 +58,25 @@ typedef struct ErrorStack {
 /// \retval USER__NULL_INPUT_ERR if myError is NULL
 int addRErrorMsg(struct ErrorStack *myError, const int status, const char *msg);
 
+/// \brief Add an error msg to the ErrorStack struct up to MAX_ERROR_MESSAGES. Allocates ErrorStack if necessary.
+///
+/// Below is an example demonstrating the usage.
+/// \code{.cpp}
+/// struct ErrorStack* p = NULL;
+/// allocate_if_necessary_and_add_rError_msg(&p, 0, "Error Message");
+/// assert(p != NULL);
+/// \endcode
+///
+/// \param[in,out] _stack the ErrorStack struct for the error msg. Will be allocated if necessary.
+/// \param[in] status the input error status.
+/// \param[in] msg the error msg string. This string will be copied to myError.
+///
+/// \returns error code
+/// \retval 0 on success
+///
+/// \since 4.3.1
+int allocate_if_necessary_and_add_rError_msg(struct ErrorStack** _stack, const int status, const char* msg);
```

Comment:
Missing leading underscores for other parameters here and in other locations.

You can remove the leading underscore from `_stack` or add one to the other parameters.